src/chains.jl

```diff
@@ -757,10 +756,24 @@ function _cat(::Val{1}, c1::Chains, args::Chains...)
                       iter = mapreduce(range, vcat, args; init=range(c1)),
                       var = nms,
                       chain = chns)
+    weights = _weights_cat(c1.weights, getproperty.(args, :weights)...)
+
+    return Chains(value, missing, c1.name_map, c1.info, weights)
+end
 
-    return Chains(value, missing, c1.name_map, c1.info)
+function _weights_cat(w::UnitWeights...) 
+    # TODO: deal with case that weights are not one-dimensional
+    T = promote_type(eltype.(w)...)
+    return UnitWeights{T}(sum(x->size(x, 1), w))
 end
 
+function _weights_cat(w::UnitWeights...) 
```

Comment:
I guess this is supposed to be
```suggestion
function _weights_cat(w::ProbabilityWeights...) 
```
?
BTW the current definitions of `_weights_cat` are problematic since they also define, and redefine, `_weights_cat()`.